src/siemens-s7_fmt_plug.c

```diff
@@ -93,6 +93,10 @@ static int valid(char *ciphertext, struct fmt_main *self)
 		goto bail;
 	if (strlen(p) != 40)
 		goto bail;
+	if ((p = strtok(NULL, "$")) == NULL)	/* Fix bug: #1090 */
+		goto bail;
+	if (strlen(p) < 40)
+		goto bail;
```

Comment:
Unless I misunderstand something it would be a lot better using `if (strlen(p) != 40)`, no?

Please note that if you change that, you should not close this and reopen another PR. Instead, you can either push another commit to the patch-1 branch (and this commit will automatically be included in this existing PR), or even better in this case, amend to this one commit (that changes the git history but it's OK for a topic branch).